src/osd/ECBackend.cc

```diff
@@ -882,7 +883,6 @@ void ECBackend::handle_sub_read(
 	bl, j->get<2>(),
 	false);
       if (r < 0) {
-	assert(0);
```

Comment:
I think we should also deal with error in OnRecoveryReadComplete and handle_recovery_read_complete.